src/sidebar/store/modules/annotations.js

```diff
@@ -71,18 +70,14 @@ function initializeAnnotation(annotation, tag) {
   return Object.assign({}, annotation, {
     // Flag indicating whether waiting for the annotation to anchor timed out.
     $anchorTimeout: false,
-    $tag: annotation.$tag || tag,
+    $tag: annotation.$tag || generateLocalId(),
```

Comment:
This change breaks the "rules" of Redux as the `ADD_ANNOTATIONS` update function is no longer pure. This in turn could cause problems with other code or middleware that relies on this assumption.